js/content/common.js

```diff
@@ -316,11 +316,7 @@ let RequestData = (function(){
 
             if (!response.ok) { throw new Error(`HTTP ${response.status} ${response.statusText} for ${response.url}`) }
 
-            if (returnJSON) {
-                return response.json();
-            } else {
-                return response.text();
-            }
+            return response[responseType]();
```

Comment:
Personally, I'd try to avoid this as much as I could :(

I'd return the actual response object and move `.json()` and `.blob()` into the `getJson()`/`getBlob()` functions.